src/interaction/InteractionManager.js

```diff
@@ -788,19 +788,19 @@ export default class InteractionManager extends EventEmitter
             interactiveParent = false;
         }
 
-        // it has a mask! Then lets hit test that before continuing..
-        if (hitTest && displayObject._mask)
+        // it has a filterArea! Same as mask but easier, its a rectangle
+        if (hitTest && displayObject.filterArea)
         {
-            if (!displayObject._mask.containsPoint(point))
+            if (!displayObject.filterArea.contains(point.x, point.y))
             {
                 hitTest = false;
             }
         }
 
-        // it has a filterArea! Same as mask but easier, its a rectangle
-        if (hitTest && displayObject.filterArea)
+        // it has a mask! Then lets hit test that before continuing if no hitArea or filterArea is present
+        if (hitTest && displayObject._mask && !displayObject.hitArea && !displayObject.filterArea)
```

Comment:
@GoodBoyDigital actually, that part _is_ the fix.  Without it, [this](https://github.com/pixijs/pixi.js/blob/ccb99cff69889c228e820969cf8e01af699c82e6/src/core/graphics/Graphics.js#L863-L866) code will be hit (for masks not using `beginFill`) and `hitTest` will be set to false, causing the relevant events to not be emitted.

Prior to the fix, a mask overrides all other hitTests.  After the fix, priority is given to the existence of `hitArea` or `filterArea` over the presence of a mask.

Does this make sense?